--- a/drivers/gpu/drm/panel/Kconfig
+++ b/drivers/gpu/drm/panel/Kconfig
@@ -44,6 +44,18 @@
 	help
 	  Say Y here to enable support for the AUO A030JTN01 320x480 3.0" panel
 	  as found in the YLM RS-97 handheld gaming console.
+
+config DRM_PANEL_BOE_AA577
+  tristate "BOE AA577 DSC cmd mode panel"
+  depends on OF
+  depends on DRM_MIPI_DSI
+  depends on BACKLIGHT_CLASS_DEVICE
+  select DRM_DISPLAY_DSC_HELPER
+  select DRM_DISPLAY_HELPER
+  help
+    Say Y here if you want to enable support for BOE AA577 panel as
+    found in OnePlus 13R / Ace 5 (giulia). The panel has a 1264x2780
+    resolution and uses 30 bit RGB per pixel.
 
 config DRM_PANEL_BOE_BF060Y8M_AJ0
 	tristate "Boe BF060Y8M-AJ0 panel"
--- a/drivers/gpu/drm/panel/Makefile
+++ b/drivers/gpu/drm/panel/Makefile
@@ -3,6 +3,7 @@
 obj-$(CONFIG_DRM_PANEL_ARM_VERSATILE) += panel-arm-versatile.o
 obj-$(CONFIG_DRM_PANEL_ASUS_Z00T_TM5P5_NT35596) += panel-asus-z00t-tm5p5-n35596.o
 obj-$(CONFIG_DRM_PANEL_AUO_A030JTN01) += panel-auo-a030jtn01.o
+obj-$(CONFIG_DRM_PANEL_BOE_AA577) += panel-boe-aa577.o
 obj-$(CONFIG_DRM_PANEL_BOE_BF060Y8M_AJ0) += panel-boe-bf060y8m-aj0.o
 obj-$(CONFIG_DRM_PANEL_BOE_HIMAX8279D) += panel-boe-himax8279d.o
 obj-$(CONFIG_DRM_PANEL_BOE_TD4320) += panel-boe-td4320.o
